### ISSUE TITLE
index.md not converting to index.html

### ISSUE THREAD

**Issue by Mirabello** [mmistakes/so-simple-theme#225]
I am having trouble adding a new category of posts because index.md is not being converted to index.html in the _site directory.

In this case, I am trying to add a buildit category in addition to the articles and blog categories. 

### I've changed config.yml

<code>
- title: Blog
  url: /blog/
- title: Theme Setup
  url: /theme-setup/
- title: BuildIt
  url: /buildit/
- title: Search
  url: /search/
</code>

###  I've also made the following changes:

* I've added added a <code>buildit</code> subdirectory under the <code> _posts</code>  directory:
  
* I've added <code>index.html</code> for the buildit category inside the _posts subdirectory

* The generated _site shows individual markdown files converted to html for the buildit posts *except* that the buildit directory index.html is not being converted from the index.md.  In fact the index.md file is what is being transferred to the _site/buildit subdirectory.

Is there anything I am not doing correctly that tells Jekyll NOT to convert the index.md to index.html in the _site directory?

Thank you for your help.

**Comment by mmistakes** [comment]
_posts directory is for posts only. Jekyll expects dated content with filenames in this format `yyyy-mm-dd-post-title.md`. That is likely why it's ignoring your index files.

Those files are considered "pages" and need to be placed in your root folder outside of _posts. 

If you move the `buildit` folder out of _posts and into the root the index.md should be generated correctly by Jekyll and show up in `_site/buildit/index.html`

**Comment by Mirabello** [comment]
Sorry.  I was reading off the Theme setup guide: "_Categories can be used to group posts into sub-folders. If you decide to rename or add categories you will need to create new category index page."_

And "_Say you want to group all your posts under blog/ instead of articles/. In your post add categories: blog to the YAML front matter, rename or duplicate articles/index.md to blog/index.md and update the for loop to limit posts to just the blog category."_

Indeed, I wanted the contents of `buildit` to also contain multiple posts of the buildit category.  I wanted the Buildit folder to be a third type of post: ie, blog, articles (the two original) and buildit, (the third category).  

Would I still need to put the <code> buildit </code> folder outside of _posts, per your response above?  Wouldn't that conflict with another `buildit` folder that contains the index.md file for the category summary page?
Thanks!

**Comment by cbira353** [comment]
here's my directory structure - [img](http://i.imgur.com/bXHWX9s.png)

**Comment by mmistakes** [comment]
@Mirabello I think the confusion maybe with how flexible Jekyll can be with how you setup your content. That and the difference between a post and a page.

If I understand you correct you want an archive page at `/buildit` that lists all of the posts in that category.

So to do that you need to create an index "page" that is in the root of your site. Since it's a page you can't place it in the `_posts` folder. So what you'll want is `<site root>/buildit/index.md`. It could also be `/buildit/index.html` if your source in that file is HTML and not MarkDown.

You also have the option of doing something like `_pages/buildit.md` and in its YAML Front Matter assigning an permalink override of `permalink: /buildit/`. I do this with [my personal site](https://github.com/mmistakes/made-mistakes-jekyll/tree/master/src/_pages), it's completely up to you on how to organize your pages. The simpliest is just placing them in the root as you won't need a `permalink` override then.

In the `_posts` folder you can have subfolders. This is just a way of grouping your content, it is completely optional. If you want you can put all the posts files directly in `_posts/` with no folders.

I'm not 100% sure if this is still the case but at one point Jekyll would assign a category to the posts depending on what subfolder they were in. So in theory `/_posts/buildit/2017-01-01-my-new-post.md` would be assigned `category: buildit` without you having to add it manually to the YAML Front Matter.

So in summary. If you want to add a "buildit" index page that lists all posts in the `buildit` category that lives at `your-domain.com/buildit/index.html` then you'd do the following:

- add "buildit" posts to `_posts/buildit/`
- copy [`<site root>/articles/index.md`](https://github.com/mmistakes/so-simple-theme/blob/master/articles/index.md) to `<site root>/buildit/index.md` and make necessary changes so it displays only `buildit` posts.
- build your site

**Comment by mmistakes** [comment]
Is your repo public somewhere so I can look at it? It's really hard to troubleshoot these sorts of things without the actual files.

I suspect there is something else at play here. Make sure you cleaned out `_posts/buildit/index.md` as well. Previously it sounded like there were "pages" placed in the posts folder that may be overwriting the one in `<site root>/buildit/index.md`.

Ultimately your issue is Jekyll related and not the theme and without the actual files I can't really troubleshoot further.

Since `/articles/index.md` and `/blog/index.md` are working, as a test rename `<site root>/buildit/` to `<site root>/foo/` and rebuild. See if you get `_site/foo/index.html`

If you do then you know for sure you have a conflicting `buildit` file somewhere, likely with a `permalink` set that clashes.

**Comment by mmistakes** [comment]
@cbira353 Think I found your issue. `<site root>/buildit/index.md` isn't valid YAML. According to your screenshot you only have `--` at line 1. It should be 3 hypens, each YAML block needs to start and end that way. That explains why it's not being processed by Jekyll

So:

```
---
layout: page
title: Buildit
category: buildit
excerpt: "blah blah blah"
search_omit: true
---
```

**Comment by cbira353** [comment]
O. M. G!!!  That was it!

So sorry to have been so sloppy.  Still learning to check myself.  

Much thanks, Michael.
